app/controllers/civil_servants/service_agreements_controller.rb

```diff
@@ -28,9 +28,18 @@ def decline
       if service.update(civil_servant_agreed: false)
         flash[:notice] = t('.successful_decline')
       else
-        flash[:error] = t('.erroneous_decline')
+        flash[:error] = format_errors(t('.erroneous_decline'), service)
       end
       redirect_to civil_servants_service_agreements_path
     end
+
+    private
+
+    # :reek:UtilityFunction
+    def format_errors(error_header, service)
+      return '' if service.valid?
```

Comment:
Why is this necessary? I'd guess that when it reached this method, the service is invalid anyways?